cot/src/common_types.rs

```diff
@@ -80,7 +88,7 @@ impl Password {
     /// let password = Password::new("password");
     /// ```
     #[must_use]
-    pub fn new<T: Into<String>>(password: T) -> Self {
+    pub fn new<T: Into<SecureString>>(password: T) -> Self {
```

Comment:
I think we should leave it as `Into<String>`. Us using `SecureString` is our implementation detail, and we shouldn't force that on users and do any required casting ourselves.